### PR TITLE
Bug fix-multiple image attachment

### PR DESCRIPTION
**Pull request by Cma-B**
### User Story
```
As an api
in order to let the users to have multiple image for their listing
we would like to provide the functionality of adding multiple image for one listing
```
## Changes proposed in this pull request:
* modifies the listing controllers to render error message for the listing without image 
* adds the test to check the functionality of adding multiple image attached to one listing

### CODE REVIEW

**Review by Alex-Bellamy — CHANGES_REQUESTED** [review]
semaphore failing

File: db/seeds.rb [hunk 1]
```diff
@@ -1,7 +1,41 @@
...
+
+user = User.create(email: 'user@mail.com', password: 'password')
+subscriber = User.create(email: 'subscriber@mail.com', password: 'password')
+
+free_listing_1 = Listing.create(
+  category: 'Parking spot',
+  lead: 'Takes less than 1 minute to get the metro',
+  scene: 'outdoor',
+  description: 'Curabitur aliquet quam id dui posuere blandit.' * 3,
+  address: 'Björcksgatan 37 B416 52 Göteborg'
```
Comment (viktorsommar) [inline db/seeds.rb]: Small thing but it seems the street number and zip code are not separated. Should be "Björcksgatan 37B, 416 52 Göteborg"

File: db/seeds.rb [hunk 2]
```diff
@@ -1,7 +1,41 @@
...
+)
+file = URI.open('https://www.quickanddirtytips.com/sites/default/files/images/12186/Parking-Lot-of-deas.jpg')
+free_listing_1.image.attach(io: file, filename: 'first.jpg')
+
+free_listing_1 = Listing.create(
+  category: 'Parking spot',
+  lead: 'Safe area and easy access',
+  scene: 'outdoor',
+  description: 'Proin eget tortor risus. Curabitur aliquet quam id dui posuere blandit.' * 3,
+  address: 'Högbergsgatan 21116 20 Stockholm'
```
Comment (viktorsommar) [inline db/seeds.rb]: Small thing but it seems the street number and zip code are not separated. Should be "Högbergsgatan 21, 116 20 Stockholm"

File: db/seeds.rb [hunk 3]
```diff
@@ -1,7 +1,41 @@
...
+)
+file = URI.open('https://www.pymnts.com/wp-content/uploads/2018/03/parkingspot.jpg')
+free_listing_2.image.attach(io: file, filename: 'second.jpg')
+
+free_listing_3 = Listing.create(
+  category: 'Parking spot',
+  lead: 'Close to the city center',
+  scene: 'outdoor',
+  description: 'Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus.' * 3,
+  address: 'Settergatan 14115 48 Stockholm'
```
Comment (viktorsommar) [inline db/seeds.rb]: Small thing but it seems the street number and zip code are not separated. Should be "Settergatan 14, 115 48 Stockholm"

File: spec/models/listing_spec.rb [hunk 4]
```diff
@@ -23,5 +23,6 @@
     it { is_expected.to validate_presence_of :address}
     it { is_expected.to validate_presence_of :description}
     it { is_expected.to validate_presence_of :price }
+
```
Comment (viktorsommar) [inline spec/models/listing_spec.rb]: extra space

File: spec/requests/api/v1/listings/create_spec.rb [hunk 5]
```diff
@@ -55,6 +63,11 @@
     it 'listing is expected to have image attached' do
       expect(Listing.last.image.attached?).to eq true
     end
+
+    it 'listing is expected to have multiple images' do
+      expect(Listing.last.image.count).to eq 2
+
```
Comment (viktorsommar) [inline spec/requests/api/v1/listings/create_spec.rb]: extra space

**Review by viktorsommar — CHANGES_REQUESTED** [review]

File: spec/models/listing_spec.rb [hunk 6]
```diff
@@ -23,5 +22,6 @@
...
```
Comment (viktorsommar) [inline spec/models/listing_spec.rb]: There is still too much space

**Review by viktorsommar — CHANGES_REQUESTED** [review]

**Review by viktorsommar — APPROVED** [review]

**Pull request merged**